### PR TITLE
Update gene info processing for druggability revamp

### PR DESCRIPTION
**Pull request by beatrizsaldana**
#### Changes
- Removed old druggability input file (syn13363443.11) from gene_info files in config and test_config 
- Removed old druggability input file (syn13363443.11) from gene_info provenance in config and test_config 
- Updated gene_metadata version number from 13 to 14 in config and test_config
- Used new druggability source file (syn64123611.1) in config, call it `pharos_classes`
- Updated druggability custom transform to keep only the pharos_class object
- Updated gene_info druggability tests and testing assets
- Updated druggability gx validation

#### Acceptance Criteria
- ADT configs incorporate the appropriate source file/provenance updates
- transform_gene_info is updated as required to produce the target data structure
- custom transform tests are added/updated as required
- GX validation is adjusted as required

[AG-863]: https://sagebionetworks.jira.com/browse/AG-863?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by BWMac** [comment]
@beatrizsaldana I think your CI is failing because of the schema update I did to the GX tables as part of #161. If you merge those changes into your branch it should work again.

File: src/agoradatatools/etl/transform/gene_info.py
Comment (BWMac) [inline src/agoradatatools/etl/transform/gene_info.py:67]: Since this results in a data structure where the column is `druggability` and inside each value is a dictionary that always has one key `pharos_class` mapped to an array of length 1 with the string class in it (unless the whole thing is null) would it make more sense to either have that column be called `pharos_class` with the single string value in it, or keep the `druggability` name and just put the `pharos_class` string in as a single value? It just seems like this doesn't need to be a nested field any longer to me.

File: src/agoradatatools/great_expectations/gx/expectations/gene_info.json
Comment (BWMac) [inline src/agoradatatools/great_expectations/gx/expectations/gene_info.json]: It looks like the reason this `mostly=0.5` needed to be added for the check to pass is because almost half of the values are entirely missing. Meaning we are going from having a `druggability` field that passed some pretty strict validation 100% of the time to one where half of our data is entirely missing. I obviously am not super familiar with the data itself and what the value add of using `pharos_class` exclusively might be, but from a pure "data" perspective this seems pretty bad... Is there a reason that this is happening that could be remedied before this change is merged?

File: src/agoradatatools/great_expectations/gx/expectations/gene_info.json
Comment (BWMac) [inline src/agoradatatools/great_expectations/gx/expectations/gene_info.json]: @jaclynbeck-sage and I had a discussion about this in #161. It seems like my machine and yours disagree on what order this list should be in for some reason. Let's try not to commit changes like this one in the future so we don't blow up our diffs unnessessarily.

File: src/agoradatatools/great_expectations/gx/json_schemas/gene_info/druggability.json
Comment (BWMac) [inline src/agoradatatools/great_expectations/gx/json_schemas/gene_info/druggability.json:8]: There seems to be a set of string values that `pharos_class` must be. We should add that as a check to ensure that we are only getting expected values

File: src/agoradatatools/great_expectations/gx/json_schemas/gene_info/druggability.json
Comment (BWMac) [inline src/agoradatatools/great_expectations/gx/json_schemas/gene_info/druggability.json]: If there is no remedy to the missing data situation/if that is expected and totally fine, an alternative to using `mostly` at the expectation suite level is to allow this check to pass on `null` values here.

**Review by BWMac — COMMENTED** [review]
My main question which could have some implications for the shape of this PR is: Does the `druggability` field need to be nested anymore? If not, this can be simplified quite a bit.

**Review by BWMac — COMMENTED** [review]
My main question which could have some implications for the shape of this PR is: Does the `druggability` field need to be nested anymore? If not, this can be simplified quite a bit.

File: src/agoradatatools/etl/transform/gene_info.py
Comment (JessterB) [reply src/agoradatatools/etl/transform/gene_info.py:67]: We are keeping the current data structure because that is the data structure that the downstream application expects. This allows us to avoid a refactor that we don't have time to take on.

File: src/agoradatatools/great_expectations/gx/expectations/gene_info.json
Comment (JessterB) [reply src/agoradatatools/great_expectations/gx/expectations/gene_info.json]: @jaclynbeck-sage Do you have any insight into why we are missing so many pharos class values?

File: src/agoradatatools/great_expectations/gx/json_schemas/gene_info/druggability.json
Comment (JessterB) [reply src/agoradatatools/great_expectations/gx/json_schemas/gene_info/druggability.json:8]: valid values are: 
- Tdark 
- Tbio
- Tchem
- Tclin

File: src/agoradatatools/great_expectations/gx/expectations/gene_info.json
Comment (jaclynbeck-sage) [reply src/agoradatatools/great_expectations/gx/expectations/gene_info.json]: Yes, this is to be expected. `gene_info` includes _all_ genes that have data in at least one of our data sets. Many of these genes don't have a known/associated protein in the Pharos or UniProt databases, so those genes get NULL values for Pharos class.

File: src/agoradatatools/great_expectations/gx/expectations/gene_info.json
Comment (jaclynbeck-sage) [inline src/agoradatatools/great_expectations/gx/expectations/gene_info.json:435]: I'm confused why this variable is the only one with "examples". What does that mean?

File: tests/test_assets/gene_info/input/pharos_classes_good_input.csv
Comment (jaclynbeck-sage) [inline tests/test_assets/gene_info/input/pharos_classes_good_input.csv:2]: Based on your expected output files, it looks like no gene in the test data set is missing a pharos class. I think it would be a good idea to test that case, by simply deleting a few genes from the pharos class input that exist in the gene_info dataset as a whole.

File: src/agoradatatools/great_expectations/gx/expectations/gene_info.json
Comment (jaclynbeck-sage) [reply src/agoradatatools/great_expectations/gx/expectations/gene_info.json]: This is so weird. I just ran the notebook on my machine and got a _third_ different ordering. I know Python lists aren't guaranteed to be in the order you put them but... wow.

File: src/agoradatatools/great_expectations/gx/expectations/gene_info.json
Comment (BWMac) [reply src/agoradatatools/great_expectations/gx/expectations/gene_info.json]: so if this is expected/accepted, the JSON schema we use to validate `druggability` should be updated to allow null values in addition to the expected data structure. See my [comment](https://github.com/Sage-Bionetworks/agora-data-tools/pull/163#discussion_r1876225025) below.

File: src/agoradatatools/great_expectations/gx/expectations/gene_info.json
Comment (BWMac) [reply src/agoradatatools/great_expectations/gx/expectations/gene_info.json:435]: The `examples` are optional to include and just allow you to have written in the JSON schema itself what the data structure should look like. AFAIK this field is not used in any actual computational validation by JSON schema at all

File: src/agoradatatools/great_expectations/gx/json_schemas/gene_info/druggability.json
Comment (JessterB) [reply src/agoradatatools/great_expectations/gx/json_schemas/gene_info/druggability.json]: @BWMac  I think using 'mostly' would report the % missing values, while allowing nulls would not. Are there any other functional considerations?

Comment (BWMac) [reply src/agoradatatools/great_expectations/gx/json_schemas/gene_info/druggability.json]: Yes, that's true. If that is desirable then I think `mostly` > allowing `null` in JSON schema for sure.

File: src/agoradatatools/great_expectations/gx/expectations/gene_info.json
Comment (jaclynbeck-sage) [reply src/agoradatatools/great_expectations/gx/expectations/gene_info.json]: @BWMac Update: Putting the values in alphabetical order in the ipynb (or calling sorted() on the original list) seems to get them to stay in alphabetical order in the json output. So maybe we should just be doing that, assuming this works on your machine too.

File: src/agoradatatools/great_expectations/gx/json_schemas/gene_info/druggability.json
Comment (JessterB) [reply src/agoradatatools/great_expectations/gx/json_schemas/gene_info/druggability.json]: In this particular case I don't need to know the %, I was just curious. If there are no other functional differences, I guess having the % is nicer than not having the %...but if the null approach is best practice I'm totally fine with that.

File: src/agoradatatools/great_expectations/gx/expectations/gene_info.json
Comment (BWMac) [reply src/agoradatatools/great_expectations/gx/expectations/gene_info.json]: ah nice! That does work for me too. Kinda annoying but I suppose it isn't too much of a pain to do this. Thanks @jaclynbeck-sage !

Comment (jaclynbeck-sage) [reply src/agoradatatools/great_expectations/gx/expectations/gene_info.json]: @beatrizsaldana would you mind adding this change to the gene_info ipynb in your PR so we don't have commits with list order issues after this PR? I think it would be good to alphabetize the possible values for pharos class too, while you're there.

File: tests/test_assets/gene_info/input/pharos_classes_good_input.csv
Comment (beatrizsaldana) [reply tests/test_assets/gene_info/input/pharos_classes_good_input.csv:2]: There are many several genes not featured in the `pharos_classes_good_input.csv` file that result in `"druggability":null` values in the output. But I can also just remove a value from the CSV and see what happens :)

Comment (jaclynbeck-sage) [reply tests/test_assets/gene_info/input/pharos_classes_good_input.csv:2]: Oh I see, I missed the places in the output where it's null because I was only looking at the diff. My bad, no changes needed here!

Comment (beatrizsaldana) [reply tests/test_assets/gene_info/input/pharos_classes_good_input.csv:2]: No worries, I actually have no idea what happens if the pharos_class column value is missing in the input, so you inspired me to check haha :)

**Comment by beatrizsaldana** [comment]
> My main question which could have some implications for the shape of this PR is: Does the `druggability` field need to be nested anymore? If not, this can be simplified quite a bit.

@BWMac according to @JessterB the field does need to be nested.

File: src/agoradatatools/great_expectations/gx/expectations/gene_info.json
Comment (beatrizsaldana) [reply src/agoradatatools/great_expectations/gx/expectations/gene_info.json]: Done!

**Review by JessterB — APPROVED** [review]
lgtm! (assuming CI passes ;)

**Review by BWMac — APPROVED** [review]
🚀

**Review by jaclynbeck-sage — APPROVED** [review]
Looks great, assuming CI passes w/ Brad's update

**Pull request merged**
